mlir/lib/Dialect/Arith/IR/ArithOps.cpp

```diff
@@ -1014,13 +1014,14 @@ OpFoldResult arith::MaxNumFOp::fold(FoldAdaptor adaptor) {
   if (getLhs() == getRhs())
     return getRhs();
 
-  // maxnumf(x, -inf) -> x
-  if (matchPattern(adaptor.getRhs(), m_NegInfFloat()))
+  // maxnumf(x, NaN) -> x
+  if (matchPattern(adaptor.getRhs(), m_NaNFloat()))
     return getLhs();
```

Comment:
Shouldn't this be maintained but updated to check that x is not NaN?

Reply:
To check that `x` is not NaN you need to have folded `x` to some constant. This is handled by the last pattern below, which computes the result of the operation if both operands are folded.